lib/Optimizer/CodeGen/WireSetsToProfileQIR.cpp

```diff
@@ -254,6 +256,16 @@ struct WireSetToProfileQIRPass
       else
         regNameMap[disc.getOperation()] = "?";
     });
+    std::optional<std::uint32_t> highestIdentity;
+    op.walk([&](quake::BorrowWireOp op) {
+      highestIdentity = highestIdentity
+                            ? std::max(*highestIdentity, op.getIdentity())
+                            : op.getIdentity();
+    });
+    if (highestIdentity)
+      op->setAttr(cudaq::opt::QIRRequiredQubitsAttrName,
```

Comment:
I think this needs to be done over the entire Module and not just this function. We may have more than 1 kernel in the Module.

Reply:
Addressed in d00b1a001. I also made it do something similar for QIR result numbers, too.